### PR TITLE
keycloak: add scim-keycloak-user-storage-spi plugin

### PR DESCRIPTION
**Pull request by s1341**
Adds the `scim-keycloak-user-storage-spi` keycloak plugin to the list of packages plugins.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/keycloak/scim-keycloak-user-storage-spi/default.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, maven
+, javaPackages
+}:
+
+javaPackages.mavenfod rec {
+  pname = "scim-keycloak-user-storage-spi";
+  version = "beta";
```
Comment (AndersonTorres) [inline pkgs/servers/keycloak/scim-keycloak-user-storage-spi/default.nix:10]: Versions are always a version string or `unstable-yyyy-mm-dd` (date when the source was produced).

File: pkgs/servers/keycloak/scim-keycloak-user-storage-spi/default.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    JAR=$(find target -iname "*.jar")
+    install -D "$JAR" "$out/$(basename $JAR)"
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/justin-stephenson/scim-keycloak-user-storage-spi";
+    description = "A third party module that extends Keycloak, allow for user storage in an external scimv2 server";
+    sourceProvenance = with sourceTypes; [
+      fromSource
+      binaryBytecode  # dependencies
```
Comment (AndersonTorres) [inline pkgs/servers/keycloak/scim-keycloak-user-storage-spi/default.nix:35]: Which dependencies?

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/servers/keycloak/scim-keycloak-user-storage-spi/default.nix [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+  mvnHash = "sha256-CK42d+Ta1/XNQWCLaje6sI+C90YvzUcteuasVkUPfCk=";
+
+  nativeBuildInputs = [
+    maven
+  ];
+
+  installPhase = ''
+    JAR=$(find target -iname "*.jar")
+    install -D "$JAR" "$out/$(basename $JAR)"
```
Comment (SuperSandro2000) [inline pkgs/servers/keycloak/scim-keycloak-user-storage-spi/default.nix:27]: The jar file probably has a stable location. Please use that instead.

File: pkgs/servers/keycloak/scim-keycloak-user-storage-spi/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/servers/keycloak/scim-keycloak-user-storage-spi/default.nix:35]: There is none in the repo and maven cannot download it.

**Comment by s1341** [comment]
Will address these tomorrow. Thanks for the review!

File: pkgs/servers/keycloak/scim-keycloak-user-storage-spi/default.nix (hunk 2 above)
Comment (s1341) [reply pkgs/servers/keycloak/scim-keycloak-user-storage-spi/default.nix]: removing.

File: pkgs/servers/keycloak/scim-keycloak-user-storage-spi/default.nix (hunk 3 above)
Comment (s1341) [reply pkgs/servers/keycloak/scim-keycloak-user-storage-spi/default.nix]: done.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
Squash the commits. Otherwise LGTM

**Comment by s1341** [comment]
Squashed.

**Comment by s1341** [comment]
Thanks!

**Pull request merged**